packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -173,8 +173,13 @@ export function handleRedirects(
 ): InternalResult | undefined {
   if (
     NextConfig.trailingSlash &&
+    !event.headers["x-nextjs-data"] &&
     !event.rawPath.endsWith("/") &&
-    !event.headers["x-nextjs-data"]
+    ![
```

Comment:
Do you need to apply this to the block below too?

Reply:
It shouldn't need the file checks, if that's what you mean, as the next block would only be triggered if `trailingSlash` is disabled.

Reply:
If there's linting issue, you can run `npm run lint:fix`

Reply:
Sorry about that. It should be fixed now.